### PR TITLE
ardour: 7.5 -> 8.0

### PR DESCRIPTION
**Pull request by ctem**
## Description of changes

This PR includes three commits, only the first of which is necessary to upgrade Ardour. The 6.9 package removal commit is for housekeeping, and the optimization commit is for convenience. If these additional commits are more appropriate in separate PRs, I'm happy to accommodate.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by magnetophon** [comment]
Thanks for the PR!

Personallly, I'd feel better if we keep 7.5 around until 8 has proven it's stability.

**Comment by olynch** [comment]
> Thanks for the PR!
> 
> Personallly, I'd feel better if we keep 7.5 around until 8 has proven it's stability.

I feel like it should not be up to the maintainers of nixpkgs to gate *application packages* on versions (as opposed to libraries that might break lots of downstream stuff). If you want to have stable application packages, you can run stable nixos, or add an ardour7 package, or use an overlay, or simply not update.

**Comment by magnetophon** [comment]
> > Thanks for the PR!
> > Personallly, I'd feel better if we keep 7.5 around until 8 has proven it's stability.
> 
> I feel like it should not be up to the maintainers of nixpkgs to gate _application packages_ on versions (as opposed to libraries that might break lots of downstream stuff). If you want to have stable application packages, you can run stable nixos, or add an ardour7 package, or use an overlay, or simply not update.

OK, fair enough.

**Review by magnetophon — APPROVED** [review]
Result of `nixpkgs-review pr 259979` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>ardour</li>
  </ul>
</details>

**Comment by ctem** [comment]
> > Thanks for the PR!
> > Personallly, I'd feel better if we keep 7.5 around until 8 has proven it's stability.
> 
> I feel like it should not be up to the maintainers of nixpkgs to gate _application packages_ on versions (as opposed to libraries that might break lots of downstream stuff). If you want to have stable application packages, you can run stable nixos, or add an ardour7 package, or use an overlay, or simply not update.

I agree, but I interpreted "keep 7.5 around" as meaning replace the temporary `ardour_6` package with a temporary `ardour_7` package as the latest fail-safe. This was certainly a good idea last time—7.0 had [issues](https://ardour.org/news/7.1.html)—so I'd be happy to amend the PR if there are no objections.

**Comment by magnetophon** [comment]
> > > Thanks for the PR!
> > > Personallly, I'd feel better if we keep 7.5 around until 8 has proven it's stability.
> > 
> > 
> > I feel like it should not be up to the maintainers of nixpkgs to gate _application packages_ on versions (as opposed to libraries that might break lots of downstream stuff). If you want to have stable application packages, you can run stable nixos, or add an ardour7 package, or use an overlay, or simply not update.
> 
> I agree, but I interpreted "keep 7.5 around" as meaning replace the temporary `ardour_6` package with a temporary `ardour_7` package as the latest fail-safe. This was certainly a good idea last time—7.0 had [issues](https://ardour.org/news/7.1.html)—so I'd be happy to amend the PR if there are no objections.

Yes, that is what I meant.

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -30621,7 +30621,6 @@ with pkgs;
     texlive = texlive.combined.scheme-medium;
   };
 
-  ardour_6 = callPackage ../applications/audio/ardour/6.nix { };
```
Comment (wegank) [inline pkgs/top-level/all-packages.nix:30624]: The deletion notice must be added to `aliases.nix`.

**Review by magnetophon — APPROVED** [review]
Result of `nixpkgs-review pr 259979` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>ardour</li>
    <li>ardour_7</li>
  </ul>
</details>

**Pull request merged**